src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/TestSuiteStepsFlow.tsx

```diff
@@ -0,0 +1,106 @@
+import {useEffect, useMemo, useState} from 'react';
+import ReactFlow, {Controls, Edge, Node} from 'reactflow';
+import 'reactflow/dist/style.css';
+
+import {nanoid} from '@reduxjs/toolkit';
+
+import {LocalStep} from '@src/models/testSuite';
+
+import IntersectionNode from './Nodes/IntersectionNode';
+import StepNode from './Nodes/StepNode';
+import {ReactFlowContainer} from './SettingsTests.styled';
+
+interface ExtendedNode extends Node {
+  group?: number | string;
+}
+
+type TestSuiteStepsFlowProps = {
+  steps: LocalStep[][];
+  setSteps: (steps: LocalStep[][]) => void;
+  showModal: (value: string, group: number | string) => void;
+};
+
+const TestSuiteStepsFlow: React.FC<TestSuiteStepsFlowProps> = props => {
+  const {steps, setSteps, showModal} = props;
+
+  const [nodes, setNodes] = useState<ExtendedNode[]>([]);
+  const [edges, setEdges] = useState<Edge[]>([]);
+
+  const removeNode = (id: string, group: number) => {
+    if (steps[group].length === 1) {
+      setSteps([...steps.slice(0, group), ...steps.slice(group + 1)]);
+    } else {
+      setSteps([...steps.slice(0, group), steps[group].filter(item => item.id !== id), ...steps.slice(group + 1)]);
+    }
+  };
+
+  useEffect(() => {
+    const newNodes: ExtendedNode[] = [];
+
+    steps.forEach((step, index) => {
```

Comment:
As in commented in previous review:
* it could be mapped instead of `forEach` `push`ed
* `position` could be built externally, as here it looks like random `x`/`y` numbers
* there could be optionally a factory function for tests/intersection objects